### PR TITLE
feat(icon): add color prop to the icon component

### PR DESCRIPTION
**Pull request by masoudmanson**
## Summary

**Icon**
GitHub ticket: #336 

The  new color prop accepts one of the following values, which are defined in the defaultTheme.ts file:

- "beta"
- "gray"
- "primary"
- "secondary"
- "info"
- "success"
- "warning"
- "error"

However, these values can be modified through the use of custom themes created by products.

## Checklist

- [x] Default Story in Storybook
- [x] LivePreview Story in Storybook
- [x] Test Story in Storybook
- [x] Tests written
- [x] Variables from `defaultTheme.ts` used wherever possible
- [x] If updating an existing component, depreciate flag has been used where necessary
- [ ] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

File: src/core/Icon/index.stories.tsx [hunk 1]
```diff
@@ -11,70 +11,71 @@ import {
...
   );
 };
 
 export default {
+  argTypes: {
+    color: {
+      control: {
+        type: "select",
+      },
+      options: [
```
Comment (tihuan) [inline src/core/Icon/index.stories.tsx:30]: love the expanded options!

File: src/core/Icon/index.stories.tsx [hunk 2]
```diff
@@ -92,10 +93,12 @@ const IconBankWrapper = styled("div")`
 `;
 
 const IconWrapper = styled("div")`
-  ${(props: CommonThemeProps) => {
+  ${(props: CommonThemeProps & { color?: SdsIconColorType }) => {
     const colors = getColors(props);
     const spacings = getSpaces(props);
 
+    const { color } = props;
```
Comment (tihuan) [inline src/core/Icon/index.stories.tsx:100]: should we default this to `primary`?

E.g.,

```ts
const { color = "primary" } = props;

...

background-color: ${colors.[color][400]};
```

File: src/core/Icon/style.ts [hunk 3]
```diff
@@ -35,19 +53,21 @@ function buttonStyle(): SerializedStyles {
 }
 
 function staticStyle<IconName extends keyof IconNameToSizes>(
-  props: IconExtraProps<IconName>
+  props: StyledSvgIconProps<IconName>
 ): SerializedStyles {
+  const { iconColor } = props;
```
Comment (tihuan) [inline src/core/Icon/style.ts:58]: same here 😄 

```ts
const { iconColor = "primary" } = props;
```

File: src/core/Icon/style.ts [hunk 4]
```diff
@@ -35,19 +53,21 @@ function buttonStyle(): SerializedStyles {
...
   const colors = getColors(props);
 
   return css`
-    color: ${colors?.primary[400]};
+    color: ${iconColor ? colors?.[iconColor][400] : colors?.primary[400]};
```
Comment (tihuan) [inline src/core/Icon/style.ts:62]: `color: ${colors?.[iconColor][400]};`

**Review by tihuan — APPROVED** [review]
LGTM! Thanks so much, @masoudmanson 🙏 Just a tiny nit 😄 🥇

File: src/core/Icon/index.stories.tsx (hunk 2 above)
Comment (masoudmanson) [reply src/core/Icon/index.stories.tsx:100]: Aahh, that's much easier 😅

File: src/core/Icon/style.ts (hunk 4 above)
Comment (masoudmanson) [reply src/core/Icon/style.ts:62]: Thanks for the recommendation 🥳🤩

**Review by clarsen-czi — CHANGES_REQUESTED** [review]
Conceptually the idea is there and works great! However I think we should provide more flexibility to folks so they can use any color from the library and aren't limited to the `-400` variants of them. See my comment in Chromatic for more details.

**Comment by masoudmanson** [comment]
> Conceptually the idea is there and works great! However I think we should provide more flexibility to folks so they can use any color from the library and aren't limited to the `-400` variants of them. See my comment in Chromatic for more details.

A new prop named `shade` has been added to the `Icon` component, which can be used to set the color shade of the icon to 100, 200, 300, 400, 500 or 600.

**Review by tihuan — APPROVED** [review]
Great idea thanks both!

**Review by clarsen-czi — APPROVED** [review]
This looks great, thanks for adding the additional functionality, @masoudmanson !!

**Pull request merged**